src/Routing/EntitySyncRoutes.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Drupal\entity_sync\Routing;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+use Symfony\Component\Routing\Route;
+
+/**
+ * Defines Routes for each entity sync operation.
+ *
+ * Loop through the entity sync configurations and create route for each
+ * operation defined.
+ */
+class EntitySyncRoutes implements ContainerInjectionInterface {
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Constructs a new EntitySyncRoutes object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The configuration factory.
+   */
+  public function __construct(
+    ConfigFactoryInterface $config_factory
+  ) {
+    $this->configFactory = $config_factory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(
+    ContainerInterface $container
+  ) {
+    return new static(
+      $container->get('config.factory')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function routes() {
+    $routes = [];
+
+    $sync_names = $this->configFactory->listAll('entity_sync.sync.');
+
+    // Loop through each entity_sync.sync configurations and create
+    // corresponding routes.
+    foreach ($this->configFactory->loadMultiple($sync_names) as $sync) {
+      $config = $sync->get($config_name);
+
+      $bundle = $config['entity']['bundle'];
+      $entity_type_id = $config['entity']['type_id'];
+
+      foreach ($config['operations'] as $operation) {
+        $operation_id = $operation['id'];
+
+        // Generate route id with the configuration id.
+        $route_id = 'entity_sync.sync.' . $config['id'];
+
+        // Generate route url based on the url path set in configuration.
+        $route_url = '/admin/sync/entities/' . $operation['url_path'];
+
+        // If the operation is list operation we use the import list form, if
+        // not we use the single import form.
+        $form_path = 'Drupal\entity_sync\Form\ImportBase';
```

Comment:
@krystalcode I'm not sure about the logic here. How would we fetch the form class?